web/models/respondent.ex

```diff
@@ -89,11 +89,11 @@ defmodule Ask.Respondent do
   end
 
   def mask_phone_number(phone_number) do
-    Enum.join([replace_numbers_by_hash(String.slice(phone_number, 0..-5)), String.slice(phone_number, -4, 4)], "")
+    Enum.join([replace_alphanumeric_by_hash(String.slice(phone_number, 0..-5)), String.slice(phone_number, -4, 4)], "")
```

Comment:
Revert changes on mask_phone_number.
Add a mask_respondent_entry that will be used at

```
/test/controllers/respondent_group_controller_test.exs
  32,53:       sample = group.sample |> Enum.map(&Respondent.mask_phone_number/1)
  55,53:       sample = group.sample |> Enum.map(&Respondent.mask_phone_number/1)
  493,31:       |> Enum.map(&Respondent.mask_phone_number/1)

/web/views/respondent_group_view.ex
  17,62:     sample = respondent_group.sample |> Enum.map(&Respondent.mask_phone_number/1)
```

Make mask_respondent_entry to return the respondentId without masking. If the input is a phone number, then delegate to mask_phone_number.

---

odly, the sample is stored unmasked in the DB. And is masked on rendering at `respondent_group_view.ex`. I would prefer for the stored sample to be masked. That way we could delegate to entries how to mask themself once and leave the stored that way. But that will come later. 

The phone numbers need to be masked to avoid revealing pii in general, although in this case the user has the number in the csv. I guess we are hiding it for demo/screenshot/recording of surveda.